src/Main.cpp

```diff
@@ -60,10 +60,98 @@ void Main::OnChannelsBox(wxCommandEvent& event) {
 	{
 		DisplayMsg(m);
 	}
+	SendText->Enable(true);
+	SendBtn->Enable(true);
+	SendText->SetFocus();
 }
 
-void Main::DisplayMsg(iMessage& m)
-{
+void Main::DisplayMsg(iMessage& m) {
 	auto msg = Message(m.timestamp, storage.GetCurrentChannel().members[m.member_id].name, wxString(m.text)).FormatToPrint();
 	ChatDisplay->AppendText(msg);
 }
+
+bool Main::PromptLogin(std::string& outUsername) 
+{
+	LoginForm* loginForm = new LoginForm(this);
+	LoginController controller(loginForm, nullptr);
+
+	if (loginForm->ShowModal() == wxID_OK) {
+		outUsername = loginForm->UsernameTextBox->GetValue().ToStdString();
+		delete loginForm;
+		return true;
+	}
+
+	delete loginForm;
```

Comment:
It is generally ill adviced to use new&delete in cpp. Make sure it necessary here.